pkg/kv/kvserver/rangefeed/scheduled_processor.go

```diff
@@ -57,9 +57,16 @@ type ScheduledProcessor struct {
 
 	requestQueue chan request
 	eventC       chan *event
-	// If true, processor is not processing data anymore and waiting for registrations
-	// to be complete.
-	stopping bool
+
+	// stopping, when true, indicates that a stop request has been processed.
```

Comment:
Add a few words about why the two snippets are not equivalent:

```go
stopping.Load()
```

```go
select {
  case <-stoppdC:
    return true
  default:
    return false
}
```

Reply:
Interestingly, while I can explain how they are subtly different in the current code, I think one of my conclusions yesterday that really, we probably could live with _only_ `stoppedC` if we wanted to because (1) once stopped is set, all registrations have been disconnected, (2) the processor won't send any more events to those registrations, and (3) all new registrations will fail. So, once stopped is set to true, any following request is effectively going to fail one way or another. So I could see an argument that we should closed stoppedC in stopInternal and then do exactly the snippet you have there.